### PR TITLE
Oldtoolset

### PR DESCRIPTION
**Pull request by moonshadow565**
Fixes for inibin parser....

### CODE REVIEW

File: ContentSerializer/ProgramInterface.cs [hunk 1]
```diff
@@ -549,6 +551,45 @@ public int ExportToMap(string target)
...
+        public void FindValueType(int type,bool filter)
+        {
+            foreach(var entry in _files)
+            {
+                var file = _manager.ReadFile(entry.FullName).Uncompress();
+                var inibin = Inibin.DeserializeInibin(file, entry.FullName);
+                foreach (var kvp in inibin.Content)
+                {
+  
+                    if(kvp.Value.Type == type)
```
Comment (MythicManiac) [inline ContentSerializer/ProgramInterface.cs]: This probably should be

``` CS
if(kvp.Value.Type != type)
{
    continue;
}
```

Lower nesting level is generally better

File: ContentSerializer/ProgramInterface.cs [hunk 2]
```diff
@@ -72,6 +73,7 @@ public void ListPaterns()
             foreach(var patern in _paterns)
```
Comment (MythicManiac) [inline ContentSerializer/ProgramInterface.cs:73]: `pattern`*

File: ContentSerializer/ProgramInterface.cs [hunk 3]
```diff
@@ -549,6 +551,45 @@ public int ExportToMap(string target)
             return 0;
         }
 
+        public void FindFiles(string section, string name)
+        {
+            UInt32 hash = HashFunctions.GetInibinHash(section, name);
```
Comment (MythicManiac) [inline ContentSerializer/ProgramInterface.cs]: `UInt32` can be declared just as `uint` AFAIK

File: ContentSerializer/ProgramInterface.cs [hunk 4]
```diff
@@ -549,6 +551,45 @@ public int ExportToMap(string target)
...
+                {
+  
+                    if(kvp.Value.Type == type)
+                    {
+                        if (_draft.Hashes.ContainsKey(kvp.Key))
+                        {
+                            var sn = _draft.Hashes[kvp.Key].First();
+                            Console.WriteLine("{0} {2} {3}: {1}", entry.FullName, kvp.Value.Value.ToString(),
+                                sn.Key, sn.Value.First());
+                        }else if (!filter) {
```
Comment (MythicManiac) [inline ContentSerializer/ProgramInterface.cs]: Move the `else if (!filter)` to a new line please, or at the very least the opening bracket.

File: ContentSerializer/ProgramInterfaceCLI.cs [hunk 5]
```diff
@@ -15,19 +15,30 @@ class ProgramInterfaceCLI
...
             {"itemmetadata", @"Data[\/\\]Items[\/\\]metadata[\/\\](.*).inibin"},
             {"spells", @"Data(|\/Characters\/([^\/]+)|\/Shared)\/Spells\/([^\/]+)\.inibin"},
-            {"characters", @"Data\/Characters\/([^\/]+)(\/\1\.inibin)"},
+            {"char", @"Data\/Characters\/([^\/]+)(\/\1\.inibin)"},
             {"talents", @"Data\/Talents\/([^\/]+)\.inibin"},
             {"skins", @"Data\/Characters\/([^\/]+)\/Skins\/([^\/]+)\/\2\.inibin"},
             {"all", @"(.*)\.inibin"}
         };
 
+        public int LoadPatern(string filename)
```
Comment (MythicManiac) [inline ContentSerializer/ProgramInterfaceCLI.cs:31]: `Pattern` not `Patern` :P

File: ContentSerializer/ProgramInterface.cs [hunk 6]
```diff
@@ -549,6 +551,45 @@ public int ExportToMap(string target)
...
+  
+                    if(kvp.Value.Type == type)
+                    {
+                        if (_draft.Hashes.ContainsKey(kvp.Key))
+                        {
+                            var sn = _draft.Hashes[kvp.Key].First();
+                            Console.WriteLine("{0} {2} {3}: {1}", entry.FullName, kvp.Value.Value.ToString(),
+                                sn.Key, sn.Value.First());
+                        }else if (!filter) { 
+                                Console.WriteLine("{0} {1}: {2}", entry.FullName,kvp.Key.ToString(), kvp.Value.Value.ToString());
```
Comment (MythicManiac) [inline ContentSerializer/ProgramInterface.cs]: 120 character limit

File: LeagueLib/Files/InibinReader.cs [hunk 7]
```diff
@@ -101,64 +101,63 @@ private bool DeserializeSegment(int type, bool skipErrors = true)
...
-            else if (type == 6) // 3x byte values - Resource bar RGB color
+            else if (type == 6) // 3x byte values *0.1f
             {
                 byte[] bytes = new byte[3];
 
                 for (int i = 0; i < count; i++)
                 {
                     _reader.BaseStream.Read(bytes, 0, bytes.Length);
-                    values[i] = new InibinValue(type, BitConverter.ToUInt32(new byte[4] { 0, bytes[0], bytes[1], bytes[2] }, 0));
+                    values[i] = new InibinValue(type, string.Format("{0} {1:} {2}", bytes[0]/10.0f, bytes[1]/10.0f, bytes[2]/10.0f));
```
Comment (MythicManiac) [inline LeagueLib/Files/InibinReader.cs]: 120 character limit

File: LeagueLib/Files/InibinReader.cs [hunk 8]
```diff
@@ -101,64 +101,63 @@ private bool DeserializeSegment(int type, bool skipErrors = true)
...
-
-                //Console.WriteLine("Tried to read 12 byte values");
                 for (int i = 0; i < count; i++)
                 {
-                    // 4 + 4 + 4 = 12
-                    _reader.ReadInt32();
-                    _reader.ReadInt32();
-                    _reader.ReadInt32();
-                    values[i] = new InibinValue(type, "NotYetImplemented");
+                    float b=_reader.ReadSingle();
```
Comment (MythicManiac) [inline LeagueLib/Files/InibinReader.cs]: `var` preferred over declaring the type
whitespace missing around `=` (applies for next two line as well)

**Review by MythicManiac — COMMENTED** [review]
This doesn't really follow our guidelines that well. Considering the nature of the ContentSerializer project, I'd say this is still mergeable. However, any changes you make towards the guidelines will be appreciated. I'll let you make the call.

File: ContentSerializer/ProgramInterfaceCLI.cs (hunk 5 above)
Comment (moonshadow565) [reply ContentSerializer/ProgramInterfaceCLI.cs:31]: There is too much of "patern".
If i changed it to pattern i will need to change it commands too and i will need to adjust to change.
Whole command line interface is just patched up anyways.
Whole CLI is just there to ease getting  hashes anyways.
Main point of this PR was parsing inibin values the correct way.
Everything else was just a bonus.

**Review by MythicManiac — APPROVED** [review]
Good enough

**Pull request merged**